### PR TITLE
Fix for wasm3 emscripten_lib build

### PR DESCRIPTION
**Pull request by o-micron**
Fix for building wasm3 with `EMSCRIPTEN_LIB`

- `PATH_MAX` wasn't defined in that case so I check if it isn't defined I defined it with `4096`
- `strnlen` is not defined in that case, an alternative would be using plain `strlen` instead

### CODE REVIEW

File: src/path_resolver.h [hunk 1]
```diff
@@ -3,6 +3,19 @@
 
 #include "uvwasi.h"
 
+/* TODO(cjihrig): PATH_MAX_BYTES shouldn't be stack allocated. On Windows, paths
```
Comment (cjihrig) [inline src/path_resolver.h:6]: This was just recently removed. I don't think it should be added back. The next time uvwasi is updated in wasm3, anything related to `PATH_MAX` should go away.

File: src/uvwasi.c [hunk 2]
```diff
@@ -1925,8 +1925,11 @@ uvwasi_errno_t uvwasi_path_readlink(uvwasi_t* uvwasi,
     return uvwasi__translate_uv_error(r);
   }
 
+#ifdef __EMSCRIPTEN__
+  len = strlen(req.ptr);
```
Comment (cjihrig) [inline src/uvwasi.c:1929]: Can you explain why this is necessary?

Comment (o-micron) [reply src/uvwasi.c:1929]: well, `wasm3` depends on `uvwasi` and while compiling `wasm3` with emscripten's `emcc` I would get an error stating that `strnlen` is not defined and I fix the problem using `strlen` instead, the difference should be (not scaning beyond the first `buf_len` bytes of the `req.ptr` string)

File: src/path_resolver.h (hunk 1 above)
Comment (o-micron) [reply src/path_resolver.h:6]: I am technically fixing an issue divided in 2 repositories `wasm3` and `uvwasi` so things got mixed up, okay so I will fix that from `wasm3`

File: src/uvwasi.c (hunk 2 above)
Comment (cjihrig) [reply src/uvwasi.c:1929]: Could this be some sort of configuration error on your end? This file (`src/uvwasi.c`) includes `string.h`, and emcc seems to support `strnlen()`. The following code:

```c
#include <stdio.h>
#include <string.h>

int main(void) {
  char* foo = "the quick brown fox jumped over the lazy dog";
  int size = strnlen(foo, 100);
  printf("size is %d\n", size);
}
```

Compiles and runs for me with the following command:

```
docker run --rm -v `pwd`:/src trzeci/emscripten emcc /src/test.c
```

Comment (o-micron) [reply src/uvwasi.c:1929]: Yeah true that it works on plain emcc example .. that must be a bad configuration from my side, closing pull request

**Pull request closed without merging**